File: scalechain-proto-codec/src/test/scala/io/scalechain/blockchain/proto/codec/walletparts/AccountCodecSpec.scala
package io.scalechain.blockchain.proto.codec.walletparts

import io.scalechain.blockchain.proto.{Account, Hash, OutPoint}
import io.scalechain.blockchain.proto.codec.{AccountCodec, OutPointCodec}
import io.scalechain.blockchain.proto.codec.primitive.CodecSuite
import io.scalechain.blockchain.proto.test.ProtoTestData

/**
  * Created by kangmo on 5/19/16.
  */
class AccountCodecSpec extends CodecSuite with ProtoTestData {

  implicit val accountCodec = AccountCodec.codec

  val account1 = Account(
    name = "kangmo"
  )

  "AccountCodec" should {

    "roundtrip" in {
      roundtrip(account1)
    }
  }
}

File: scalechain-proto-codec/src/test/kotlin/io/scalechain/blockchain/proto/codec/walletparts/AccountCodecSpec.kt
package io.scalechain.blockchain.proto.codec.walletparts

import io.kotlintest.KTestJUnitRunner
import io.kotlintest.matchers.Matchers
import io.kotlintest.specs.FlatSpec
import io.scalechain.blockchain.proto.*
import io.scalechain.blockchain.proto.codec.CodecTestUtil

import io.scalechain.blockchain.proto.codec.AccountCodec
import org.junit.runner.RunWith

/**
  * Created by kangmo on 5/19/16.
  */
@RunWith(KTestJUnitRunner::class)
class AccountCodecSpec : FlatSpec(), Matchers, CodecTestUtil {

  val account1 = Account(
    name = "kangmo"
  )

  init {
    "AccountCodec" should "roundtrip" {
      roundTrip(AccountCodec, account1) shouldBe true
    }
  }
}

